File: src/CandidatosCore.Core/Fases/Contratado.cs
namespace CandidatosCore.Core.Fases;

public sealed class Contratado : FaseBase
{
    public override string Nombre => "Contratado";

    public override FaseBase Avanzar() =>
        throw new InvalidOperationException($"Transición inválida: {Nombre} no tiene una siguiente etapa");

    public override void Rechazar() =>
        throw new InvalidOperationException($"Transición inválida: {Nombre} -> Rechazado");
}

File: src-ts/CandidatosCore.Core/Fases/Contratado.ts
import { FaseBase } from "./FaseBase";

export class Contratado extends FaseBase {
    public get nombre(): string {
        return "Contratado";
    }

    public override avanzar(): FaseBase {
        throw new Error(`Transición inválida: ${this.nombre} no tiene una siguiente etapa`);
    }

    public override rechazar(): void {
        throw new Error(`Transición inválida: ${this.nombre} -> Rechazado`);
    }
}
